db/migrate/20190214210719_update_metadata_fields.rb

```diff
@@ -0,0 +1,184 @@
+      options: ["Yes", "No"],
+      force_options: 1,
+      is_core: 1,
+      is_default: 1,
+      group: "Sample",
+      host_genomes: HostGenome.all
+    )
+
+    to_create << MetadataField.new(
+      name: "isolate",
+      display_name: "Isolate",
+      description: "Whether or not the sample is an isolate",
+      base_type: Metadatum::STRING_TYPE,
+      options: ["Yes", "No"],
+      force_options: 1,
+      is_core: 1,
+      is_default: 1,
+      group: "Sample",
+      host_genomes: HostGenome.all
+    )
+
+    if HostGenome.find_by(name: "Mosquito")
+      to_create << MetadataField.new(
+        name: "gravid",
+        display_name: "Gravid",
+        description: "Whether or not the host(s) were gravid",
+        base_type: Metadatum::STRING_TYPE,
+        options: ["Yes", "No", "Mixed", "Unknown"],
+        force_options: 1,
+        is_core: 1,
+        is_default: 1,
+        group: "Host",
+        host_genomes: [HostGenome.find_by(name: "Mosquito")]
+      )
+    end
+
+    mf = MetadataField.new(
+      name: "diseases_and_conditions",
+      display_name: "Diseases and Conditions",
+      description: "Diseases or conditions observed in the host",
+      base_type: Metadatum::STRING_TYPE,
+      is_core: 1,
+      is_default: 1,
+      group: "Host"
+    )
+    ["Mouse", "Cat", "Pig"].each do |h|
+      mf.host_genomes << HostGenome.find_by(name: h) if HostGenome.find_by(name: h)
+    end
+    to_create << mf
+
+    # Create the fields unless they already exist
+    to_create.each do |m|
+      m.save unless MetadataField.find_by(name: m.name)
+    end
+
+    #########################
+    # Modify existing fields
+    #########################
+
+    # Replace 'participant' with 'host' in the description
+    MetadataField.find_each { |field| field.update(description: field.description.gsub("participant", "host")) }
+
+    mf = MetadataField.find_by(name: "participant_id")
+    mf.update(name: "host_id", display_name: "Host ID") if mf
+
+    mf = MetadataField.find_by(name: "blood_fed")
+    hg = HostGenome.find_by(name: "Tick")
+    mf.host_genomes << hg if mf && hg
+
+    mf = MetadataField.find_by(name: "life_stage")
+    mf.update(name: "host_life_stage", display_name: "Host Life Stage") if mf
+
+    mf = MetadataField.find_by(name: "age")
+    mf.update(name: "host_age", display_name: "Host Age", description: "Age of the host (default in years)") if mf
+
+    mf = MetadataField.find_by(name: "sex")
+    mf.update(name: "host_sex", display_name: "Host Sex", description: "Sex of the host") if mf
+
+    mf = MetadataField.find_by(name: "genus_species")
+    mf.update(name: "host_genus_species", display_name: "Host Genus Species") if mf
+
+    # Assign all the fields on Cat to Pig (non-human defaults)
+    if HostGenome.find_by(name: "Pig") && HostGenome.find_by(name: "Cat")
+      HostGenome.find_by(name: "Pig").metadata_fields << HostGenome.find_by(name: "Cat").metadata_fields
```

Comment:
Can you do `HostGenome.find_by(name: "Pig").metadata_fields << (HostGenome.find_by(name: "Cat").metadata_fields - HostGenome.find_by(name: "Pig").metadata_fields)` just to be safe?

If Pig already has some of Cat's metadata_fields, I think this will throw an error.

I actually already added Cat's metadata_fields to Pig in prod. Happy to remove (and happy to do a migration next time if we think that's best)

Reply:
👍